src/Transport/Http/HttpAsyncClient.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * This file is part of the prooph/event-store-client.
+ * (c) 2018-2018 prooph software GmbH <contact@prooph.de>
+ * (c) 2018-2018 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStoreClient\Transport\Http;
+
+use Amp\Artax\Client;
+use Amp\Artax\DefaultClient;
+use Amp\Artax\Request;
+use Amp\Artax\Response;
+use Prooph\EventStoreClient\UserCredentials;
+use Throwable;
+
+/** @internal  */
+class HttpAsyncClient
+{
+    /** @var Client */
+    private $client;
+
+    public function __construct(int $operationTimeout)
+    {
+        $this->client = new DefaultClient();
```

Comment:
Instantiation in the constructor not best practice, would make unit testing this class difficult. Why not inject the client?

Reply:
Or provide `DefaultClient` only as default if no other is injected as argument.

Reply:
This is not a dependency but an implementation detail.